auth/domain/src/main/java/com/nolawiworkineh/auth/domain/UserDataValidator.kt

```diff
@@ -0,0 +1,31 @@
+package com.nolawiworkineh.auth.domain
+
+class UserDataValidator(private val patternValidator: PatternValidator) {
+
+    fun isValidName(name: String): Boolean {
+        return patternValidator.matches(name.trim())
+    }
+
+    fun isValidEmail(email: String): Boolean {
+        return patternValidator.matches(email.trim())
+    }
+
+    fun validatePassword(password: String): PasswordValidationState {
+        val trimmedPassword = password.trim()
```

Comment:
Be careful when trimming passwords; spaces can sometimes be considered valid characters in a password